packages/testing/src/assert-to-failure-interceptor.ts

```diff
@@ -0,0 +1,38 @@
+import assert from 'assert';
+import { ApplicationFailure, WorkflowInterceptors } from '@temporalio/workflow';
+
+/**
+ * Simple interceptor that transforms assertion errors into non retryable failures.
```

Comment:
Worth specifying you mean `assert.AssertionError`s specifically

Reply:
Sure